mlterm: darwin fixes

Atemu opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Atemu committed:
mlterm: unbreak darwin

Atemu committed:
mlterm: enable libvte on darwin

Atemu committed:
mlterm: support all platforms

This app can run on pretty much anything. Windows, BSDs, Solaris, Android,
Haiku; you name it.

wegank commented on pkgs/applications/terminal-emulators/mlterm/default.nix:
```diff
@@ -28,20 +28,20 @@ stdenv.mkDerivation rec {
     ibus
-  ] ++ lib.optionals (stdenv.system != "aarch64-linux") [
-    # FIXME Currently broken on aarch64-linux
+  ] ++ lib.optionals (!stdenv.isAarch64 && !stdenv.isDarwin) [
```
We could just update `uim.meta.broken` and then

```suggestion
  ] ++ lib.optionals (!uim.meta.broken) [
```

wegank replied:
Also, `uim` actually compiles on `aarch64-linux`...

wegank reviewed (approved):
Just a squash away.

Atemu committed:
mlterm: clean up libssh support

openssl is not necessary as a direct dep and we no longer need the linker magic.

Atemu committed:
mlterm: remove vrthra from maintainers

Atemu commented:
@ramkromberg do you still maintain mlterm?

Atemu commented:
Thanks a bunch @wegank!

7c6f434c commented:
… and now `mlterm` with `libssh2` breaks without `openssl` again. I want to readd it and write something about its removal not being a good idea, how to word it? (Or should we just disable libssh2 by default?)

Atemu commented:
I'm a bit confused what you're trying to say. Could you reword that?

Does libssh2 get fixed by the PR making openssl a propagated dep of libssh2?

7c6f434c commented:
https://hydra.nixos.org/build/223620190

I have looked at the code; as mlterm uses openssl directly, I think it is reasonable to list openssl as a dependency. (And mlterm is listed as still-failing in the libssh2 PR)
